frame/pools/src/lib.rs

```diff
@@ -0,0 +1,1135 @@
+			// does not yet exist.
+			{
+				let mut unbond_pool =
+					sub_pools.with_era.entry(current_era).or_insert_with(|| UnbondPool::default());
+				let points_to_issue = unbond_pool.points_to_issue(balance_to_unbond);
+				unbond_pool.points = unbond_pool.points.saturating_add(points_to_issue);
+				unbond_pool.balance = unbond_pool.balance.saturating_add(balance_to_unbond);
+			}
+
+			delegator.unbonding_era = Some(current_era);
+
+			Self::deposit_event(Event::<T>::Unbonded {
+				delegator: who.clone(),
+				pool: delegator.pool.clone(),
+				amount: balance_to_unbond,
+			});
+
+			// Now that we know everything has worked write the items to storage.
+			BondedPools::insert(&delegator.pool, bonded_pool);
+			SubPoolsStorage::insert(&delegator.pool, sub_pools);
+			Delegators::insert(who, delegator);
+
+			Ok(())
+		}
+
+		#[pallet::weight(666)]
+		pub fn withdraw_unbonded(origin: OriginFor<T>, num_slashing_spans: u32) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let delegator = Delegators::<T>::get(&who).ok_or(Error::<T>::DelegatorNotFound)?;
+
+			let unbonding_era = delegator.unbonding_era.ok_or(Error::<T>::NotUnbonding)?;
+			let current_era = T::StakingInterface::current_era().unwrap_or(Zero::zero());
+			if current_era.saturating_sub(unbonding_era) < T::StakingInterface::bonding_duration() {
+				return Err(Error::<T>::NotUnbondedYet.into())
+			};
+
+			let mut sub_pools =
+				SubPoolsStorage::<T>::get(&delegator.pool).ok_or(Error::<T>::SubPoolsNotFound)?;
+
+			let balance_to_unbond = if let Some(pool) = sub_pools.with_era.get_mut(&unbonding_era) {
+				let balance_to_unbond = pool.balance_to_unbond(delegator.points);
+				pool.points = pool.points.saturating_sub(delegator.points);
+				pool.balance = pool.balance.saturating_sub(balance_to_unbond);
+
+				balance_to_unbond
+			} else {
+				// A pool does not belong to this era, so it must have been merged to the era-less
+				// pool.
+				let balance_to_unbond = sub_pools.no_era.balance_to_unbond(delegator.points);
+				sub_pools.no_era.points = sub_pools.no_era.points.saturating_sub(delegator.points);
+				sub_pools.no_era.balance =
+					sub_pools.no_era.balance.saturating_sub(balance_to_unbond);
+
+				balance_to_unbond
+			};
+
+			T::StakingInterface::withdraw_unbonded(delegator.pool.clone(), num_slashing_spans)?;
+
+			T::Currency::transfer(
+				&delegator.pool,
+				&who,
+				balance_to_unbond,
+				ExistenceRequirement::AllowDeath,
+			)
+			.map_err(|e| {
+				log::warn!("system logic error: pool could not withdraw_unbonded due to lack of free balance");
+				e
+			})?;
+
+			SubPoolsStorage::<T>::insert(&delegator.pool, sub_pools);
+			Delegators::<T>::remove(&who);
+
+			Self::deposit_event(Event::<T>::Withdrawn {
+				delegator: who,
+				pool: delegator.pool,
+				amount: balance_to_unbond,
+			});
+
+			Ok(())
+		}
+
+		// Create a pool.
+		//
+		// * `targets`: _Stash_ addresses of the validators to nominate
```

Comment:
the term target is used in the code to reference multiple things. I recommend removing all of them. This should be called `validators` (since this pallet assumes the existence of a staking system and can use its terminology).

In other places, it seems to refer to pool's account id. I would call those `pool_account`.

Reply:
63377e106555c7d681ad434c0ed4cfcfad0d498b